src/login/lobby.cpp

```diff
@@ -779,42 +781,78 @@ int32 lobbyview_parse(int32 fd)
                     std::memset(CharName, 0, sizeof(CharName));
                     std::memcpy(CharName, sessions[fd]->rdata.data() + 32, sizeof(CharName));
 
-                    // find assigns
-                    const char* fmtQuery = "SELECT charname FROM chars WHERE charname LIKE '%s'";
+                    // Sanitize name
+                    char escapedCharName[16 * 2 + 1];
+                    sql->EscapeString(escapedCharName, CharName);
+
+                    std::optional<std::string> invalidNameReason = std::nullopt;
 
-                    std::string myNameIs(&CharName[0]);
-                    bool        invalidName = false;
-                    for (auto letters : myNameIs)
+                    // Check for invalid characters
+                    std::string nameStr(&escapedCharName[0]);
+                    for (auto letters : nameStr)
                     {
                         if (!std::isalpha(letters))
                         {
-                            invalidName = true;
+                            invalidNameReason = "Invalid characters present in name.";
                             break;
                         }
                     }
 
-                    char escapedCharName[16 * 2 + 1];
-                    sql->EscapeString(escapedCharName, CharName);
-                    if (sql->Query(fmtQuery, escapedCharName) == SQL_ERROR)
+                    // Check if the name is already in use by another character
+                    if (sql->Query( "SELECT charname FROM chars WHERE charname LIKE '%s'", escapedCharName) == SQL_ERROR)
                     {
-                        do_close_lobbyview(sd, fd);
-                        return -1;
+                        invalidNameReason = "Internal entity name query failed.";
+                    }
+                    else if (sql->NumRows() != 0)
+                    {
+                        invalidNameReason = "Name already in use.";
                     }
 
-                    if (sql->NumRows() != 0 || invalidName)
+                    // (optional) Check if the name is in use by NPC or Mob entities
```

Comment:
Retail allows this. I have met many players with names identical to a monster or NPC.

Reply:
I see you made it a setting, just making sure everyone realizes that's a thing.